File: projects/allee_effect/code/allee_effect_script.py
import matplotlib.pyplot as plt
import numpy as np


r = 0.5
k = 200
a_strong = 60
n = np.linspace(0, k * 1.05, 5000)

growth_no = r * n * (1 - n / k)
growth_weak = (r / k) * n ** 2 * (1 - n / k)
growth_strong = r * n * (1 - n / k) * (n / a_strong - 1)

plt.plot(n, growth_no, linewidth=2, label="No Allee Effect")
plt.plot(n, growth_weak, linewidth=2, label="Weak Allee Effect")
plt.plot(n, growth_strong, linewidth=2, label="Strong Allee Effect")
plt.axhline(0, color="black", linewidth=1.5)
plt.title("Population Growth Models with Allee Effects")
plt.xlabel("Population Density")
plt.ylabel("dN/dt")
plt.xlim(0, k * 1.1)
plt.legend(loc="lower right")
plt.show()

File: projects/allee_effect/code/allee_effect_script.m
%--- Model parameters and density vector
r = 0.5; % Intrinsic growth rate
K = 200; % Carrying capacity
A_strong = 60; % Strong effect threshold
N = linspace(0, K*1.05, 5000);
%--- Compute growth curves
growth_no = r * N .* (1 - N/K);
growth_weak = (r/K) * N.^2 .* (1 - N/K);
growth_strong = r * N .* (1 - N/K) .* (N/A_strong - 1);
%--- Plot all three curves
figure;
hold on
plot(N, growth_no, 'LineWidth', 2);
plot(N, growth_weak, 'LineWidth', 2);
plot(N, growth_strong, 'LineWidth', 2);
yline(0, 'k-', 'LineWidth', 1.5); 
title('Population Growth Models with Allee Effects');
xlabel('Population Density'); ylabel('dN/dt');
legend('No Allee Effect', 'Weak Allee Effect', 'Strong Allee Effect', 'Location', 'southeast');
xlim([0, K*1.1])
hold off
